src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,96 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final int EMPTY_SIZE = 0;
+    private static final int CAPACITY_DIVISOR = 2;
+    private Object[] dataElement = new Object[DEFAULT_CAPACITY];
```

Comment:
```suggestion
    private T[] elements;
```

it is not necessary to create an Object immediately. And fix the relevant errors related to this